src/commands/exec.ts

```diff
@@ -253,13 +260,58 @@ export default class Exec extends BaseCommand {
     ws_stream.write(buffer);
   }
 
+  getServiceReplica(replica_flag: string, replicas: Replica[]): Replica {
+    const resources: Dictionary<any> = {};
+    for (const rep of replicas) {
+      const { resource_name } = ResourceSlugUtils.parse(rep.resource_ref);
+      if (resources.hasOwnProperty(resource_name)) {
+        resources[resource_name].push(rep);
+      } else {
+        resources[resource_name] = [rep];
+      }
+    }
+    const resource_names = Object.keys(resources);
+
+    let service_replicas = [];
+    let replica_idx;
+    if (new RegExp('^[0-9]+$').test(replica_flag)) {
+      if (resource_names.length > 1) {
+        throw new ArchitectError(`More than one service is found [${resource_names.join(', ')}]. Please specify the replica in the form of <service-name>:<replica-index>.`);
+      }
+
+      service_replicas = replicas;
+      replica_idx = parseInt(replica_flag);
+    } else if (new RegExp('^[a-z]+[a-zA-Z0-9_-]+:[0-9]+$').test(replica_flag)) {
```

Comment:
This can be simplified a little and not require using regular expressions to check the input
```typescript
replica_parts = replica_flag.split(':');
if (replica_parts.length === 1 && !isNaN(replica_parts[0])) {
    // same code currently there, replica_idx = parseInt(replica_parts[0]) is okay
} else if (replica_parts.length == 2) {
    // same code you've got - already did the split
    const [service_name, service_replica_idx] = replica_parts;
}
```

The other error handling you've got for making sure the service name exists and that the replica_idx is within the bounds of available replicas should handle any weird inputs somebody wants to throw

Reply:
(The main advantage to avoiding a regex on the service name especially is that if we changed the regex for what is allowed in our architect.yml schema, we can very easily forget to change this code - so just accepting any string before the `:` and just checking against our set of service names avoids that potential future bug)

Reply:
That is a good point. I'll fix that.